--- a/src/main.cpp	2016-04-16 07:34:22.984000000 +0200
+++ b/src/main.cpp	2016-04-16 08:31:38.000000000 +0200
@@ -75,7 +75,7 @@
 }
 
 /////////////////////////////////////////////////////////////////////////////////////////
-QStringList g_qtFlags = {"--enable-viewport", "--enable-viewport-meta", "--disable-gpu"};
+QStringList g_qtFlags = {"--enable-viewport", "--enable-viewport-meta"};
 
 /////////////////////////////////////////////////////////////////////////////////////////
 int main(int argc, char *argv[])
