--- a/qtbase/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmscreen.cpp	2017-03-22 08:46:05.841653089 +0100
+++ b/qtbase/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmscreen.cpp	2017-03-22 08:46:27.848546329 +0100
@@ -80,7 +80,7 @@

     QScopedPointer<FrameBuffer> fb(new FrameBuffer);

-    int ret = drmModeAddFB(device()->fd(), width, height, 24, 32,
+    int ret = drmModeAddFB(device()->fd(), width, height, 32, 32,
                            stride, handle, &fb->fb);

     if (ret) {
@@ -131,7 +131,7 @@
         m_gbm_surface = gbm_surface_create(static_cast<QEglFSKmsGbmDevice *>(device())->gbmDevice(),
                                            rawGeometry().width(),
                                            rawGeometry().height(),
-                                           GBM_FORMAT_XRGB8888,
+                                           GBM_FORMAT_ARGB8888,
                                            GBM_BO_USE_SCANOUT | GBM_BO_USE_RENDERING);
     }
     return m_gbm_surface;
