--- a/qtbase/configure	2016-03-03 12:35:22.000000000 +0100
+++ b/qtbase/configure	2016-04-21 13:10:04.984000000 +0200
@@ -5540,6 +5540,11 @@
 
         if [ "$CFG_XCB" = "qt" ]; then
             QT_CONFIG="$QT_CONFIG xcb-qt"
+
+	    if compileTest qpa/xcb-glx "xcb-glx" $QMAKE_CFLAGS_XCB $QMAKE_LIBS_XCB; then
+              CFG_XCB_GLX=yes
+              QT_CONFIG="$QT_CONFIG xcb-glx"
+            fi
         else
             CFG_XCB="system"
             if [ -n "$PKG_CONFIG" ] && $PKG_CONFIG --exists "xcb >= 1.5" 2>/dev/null; then
